### PR TITLE
Fix ExternalTaskSensor cant check zipped dag

### LINKED ISSUE #14264: AirflowException: The external DAG was deleted when external_dag_id references zipped DAG
**Apache Airflow version**: 2.0.0

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`): v1.16.3

**Environment**:

- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release): Debian GNU/Linux 10 (buster)
- **Kernel** (e.g. `uname -a`): Linux airflow 3.10.0-1127.10.1.el7.x86_64 #1 SMP Tue May 26 15:05:43 EDT 2020 x86_64 GNU/Linux
- **Install tools**:
- **Others**:

**What happened**:

ExternalTaskSensor with check_existence=True referencing an external DAG inside a .zip file raises the following exception:
```
ERROR - The external DAG dag_a /opt/airflow-data/dags/my_dags.zip/dag_a.py was deleted.
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1086, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1260, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/taskinstance.py", line 1300, in _execute_task
    result = task_copy.execute(context=context)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/sensors/base.py", line 228, in execute
    while not self.poke(context):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/utils/session.py", line 65, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/sensors/external_task.py", line 159, in poke
    self._check_for_existence(session=session)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/sensors/external_task.py", line 184, in _check_for_existence
    raise AirflowException(f'The external DAG {self.external_dag_id} was deleted.')
airflow.exceptions.AirflowException: The external DAG dag_a /opt/airflow-data/dags/my_dags.zip/dag_a.py was deleted.
```

**What you expected to happen**:

The existence check should PASS.

**How to reproduce it**:

1. Create a file *dag_a.py* with the following contents:
```
from airflow import DAG
from airflow.operators.dummy import DummyOperator
from airflow.utils.timezone import datetime

DEFAULT_DATE = datetime(2015, 1, 1)
with DAG("dag_a", start_date=DEFAULT_DATE, schedule_interval="@daily") as dag:
    task_a = DummyOperator(task_id="task_a", dag=dag)
```

2. Create a file *dag_b.py* with contents:
```
from airflow import DAG
from airflow.operators.dummy import DummyOperator
from airflow.sensors.external_task import ExternalTaskSensor
from airflow.utils.timezone import datetime

DEFAULT_DATE = datetime(2015, 1, 1)

with DAG("dag_b", start_date=DEFAULT_DATE, schedule_interval="@daily") as dag:
    sense_a = ExternalTaskSensor(
        task_id="sense_a",
        external_dag_id="dag_a",
        external_task_id="task_a",
        check_existence=True
    )
    task_b = DummyOperator(task_id="task_b", dag=dag)
    sense_a >> task_b
```

3. `zip my_dags.zip dag_a.py dag_b.py`
4. Load *my_dags.zip* into airflow and run *dag_b*
5. Task *sense_a* will fail with exception above.

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by chenglongyan**
closes: #14264

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/sensors/external_task.py [hunk 1]
```diff
@@ -261,7 +262,7 @@ def _check_for_existence(self, session) -> None:
         if not dag_to_wait:
             raise AirflowException(f'The external DAG {self.external_dag_id} does not exist.')
 
-        if not os.path.exists(dag_to_wait.fileloc):
+        if not os.path.exists(correct_maybe_zipped(dag_to_wait.fileloc)):
```
Comment (ephraimbuddy) [inline airflow/sensors/external_task.py:266]: Add some tests to verify this

Comment (chenglongyan) [reply airflow/sensors/external_task.py:266]: Added.

File: tests/sensors/test_external_task_sensor.py [hunk 2]
```diff
@@ -576,6 +576,19 @@ def test_external_task_sensor_waits_for_dag_check_existence(self):
...
+    with dag_maker(dag_id="external_task_sensor_dag"):
+        op = ExternalTaskSensor(
+            task_id="external_task_sensor",
+            external_dag_id="test_zip_dag",
+            external_task_id="dummy",
+            check_existence=True
+        )
+
+    with create_session() as session:
+        op._check_for_existence(session)
```
Comment (ephraimbuddy) [inline tests/sensors/test_external_task_sensor.py]: We will need a real zipped dag to test this. You can create one at `tests/dags`

Comment (uranusjr) [reply tests/sensors/test_external_task_sensor.py]: It’s better to use a fixture to dynamically create/delete a zip file when tests are run and instead; a static zip file in `tests/dags` is opaque and easy to miss when we do refactoring. (IIRC this happened a while ago.)

See tests in `tests/models/test_dagbag.py` for some examples for this.

**Review by uranusjr — APPROVED** [review]

**Comment by potiuk** [comment]
Static checks are failing.

File: tests/sensors/test_external_task_sensor.py [hunk 3]
```diff
@@ -576,6 +610,14 @@ def test_external_task_sensor_waits_for_dag_check_existence(self):
             op.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE, ignore_ti_state=True)
 
 
+def test_external_task_sensor_check_zipped_dag_existence(dag_zip_maker):
+    with dag_zip_maker("test_external_task_sensor_check_existense.py") as dagbag:
+        with create_session() as session:
+            dag = dagbag.dags['test_external_task_sensor_check_existence']
+            op = dag.tasks[0]
+            op._check_for_existence(session)
```
Comment (uranusjr) [inline tests/sensors/test_external_task_sensor.py:642]: Did you miss an `assert` here?

Comment (chenglongyan) [reply tests/sensors/test_external_task_sensor.py:642]: _check_for_existence will raise AirflowException when external DAG does not exist.

**Comment by uranusjr** [comment]
Needs to resolve conflicts.

File: tests/sensors/test_external_task_sensor.py [hunk 4]
```diff
@@ -53,6 +58,35 @@ def clean_db():
     clear_db_runs()
 
 
+@pytest.fixture
```
Comment (potiuk) [inline tests/sensors/test_external_task_sensor.py:61]: Nice one !

**Review by potiuk — APPROVED** [review]

**Pull request merged**